streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/PassportController.java

```diff
@@ -102,6 +109,46 @@ public RestResponse signin(
         return new RestResponse().data(userInfo);
     }
 
+    @PostMapping("signinbycasdoor")
+    public RestResponse signinByCasdoor(@NotBlank(message = "{required}") String code,
+                                        @NotBlank(message = "{required}") String state) throws Exception {
+        String token = "";
+        String username = "";
+        String password = "";
+        token = casdoorAuthService.getOAuthToken(code, state);
+        CasdoorUser casdoorUser = casdoorAuthService.parseJwtToken(token);
+        User user = userService.findByName(casdoorUser.getName());
+        if (user == null) {
+            User user2 = new User();
+            user2.setUsername(casdoorUser.getName());
+            user2.setNickName(casdoorUser.getDisplayName());
+            user2.setPassword(casdoorUser.getPassword());
+            user2.setUserType(UserType.ADMIN);
```

Comment:
Set the user type is admin, this operation needs to be discussed.

Reply:
hi jakiuncle:
thanks for your contribution, looks very good, But some details need to be discussed, welcome to subscribe to our email: [dev@streampark.apache.org](mailto:dev@streampark.apache.org), discuss together.